### PR TITLE
Add drop-down menu on k8s L2 dashboards

### LINKED ISSUE #43: Add basic drop-down menu on k8s L2 dashboards
Add drop-down menu:
- Deployments
- StatefulSets
- DaemonSets
- Pods
- Containers
- Jobs
- PVCs

### PR DESCRIPTION
**Pull request by michal-gubricky**
closes #43 

Signed-off-by: michal.gubricky <michal.gubricky@ifne.eu>

### CODE REVIEW

File: jsonnet/dashboards/k8s/container-overview.libsonnet [hunk 1]
```diff
@@ -61,6 +61,42 @@ local getNextIndex(arrays) =
...
+          includeAll=true,
+          multi=true,
+        );
+
+      local podTemplate =
+        template.new(
+          name='pod',
+          label='Pod',
+          datasource='$datasource',
+          query='label_values(kube_pod_container_info{cluster=~"$cluster", namespace=~"$namespace", container=~"$container"}, pod)',
```
Comment (davidsuba) [inline jsonnet/dashboards/k8s/container-overview.libsonnet:81]: I would switch `podTemplate` and `containerTemplate` and change queries accordingly. Like in [container-detail](https://github.com/dNationCloud/kubernetes-monitoring/blob/main/jsonnet/dashboards/k8s/container-detail.libsonnet#L188)

File: jsonnet/dashboards/k8s/container-overview.libsonnet [hunk 2]
```diff
@@ -79,10 +115,11 @@ local getNextIndex(arrays) =
       local valueMaps = std.flattenArrays([valueMapsOk, valueMapsWaitingErrors, valueMapsTerminatedErrors]);
 
       local okQueries = [
-        'sum by (container, namespace, pod) (kube_pod_container_status_terminated_reason{cluster=~"$cluster", reason="Completed"} * 1)',
+        'sum by (container, namespace, pod) (kube_pod_container_status_terminated_reason{cluster=~"$cluster", pod=~"$pod", reason="Completed"} * 1)',
```
Comment (davidsuba) [inline jsonnet/dashboards/k8s/container-overview.libsonnet:118]: I don't understand why, there is only `pod=~"$pod"` and not also `container=~"$container"` and `namespace=~"$namespace"`, like in [container detail](https://github.com/dNationCloud/kubernetes-monitoring/blob/main/jsonnet/dashboards/k8s/container-detail.libsonnet#L40). And maybe it should be added to all parts of `statusExpr`.

File: jsonnet/dashboards/k8s/job-overview.libsonnet [hunk 3]
```diff
@@ -42,6 +42,30 @@ local table = grafana.tablePanel;
           hide='variable',
         );
 
+      local namespaceTemplate =
+        template.new(
+          name='namespace',
+          label='Namespace',
+          datasource='$datasource',
+          query='label_values(kube_job_info{cluster=~"$cluster", job=~"$job"}, namespace)',
```
Comment (davidsuba) [inline jsonnet/dashboards/k8s/job-overview.libsonnet:50]: I would remove `job` from this template and add `namespace` to `jobTemplate`.

File: jsonnet/dashboards/k8s/pod-overview.libsonnet [hunk 4]
```diff
@@ -65,11 +89,11 @@ local row = grafana.row;
...
-            sum by (namespace, pod) (kube_pod_status_phase{cluster=~"$cluster", phase="Running"} * 1) +
-            sum by (namespace, pod) (kube_pod_status_phase{cluster=~"$cluster", phase="Succeeded"} * 2) +
-            sum by (namespace, pod) (kube_pod_status_phase{cluster=~"$cluster", phase="Unknown"} * 3) +
-            sum by (namespace, pod) (kube_pod_status_phase{cluster=~"$cluster", phase="Failed"} * 4) +
-            sum by (namespace, pod) (kube_pod_status_phase{cluster=~"$cluster", phase="Pending"} * 5)
+            sum by (namespace, pod) (kube_pod_status_phase{cluster=~"$cluster", namespace=~"$namespace", phase="Running"} * 1) +
+            sum by (namespace, pod) (kube_pod_status_phase{cluster=~"$cluster", namespace=~"$namespace", phase="Succeeded"} * 2) +
+            sum by (namespace, pod) (kube_pod_status_phase{cluster=~"$cluster", namespace=~"$namespace", phase="Unknown"} * 3) +
+            sum by (namespace, pod) (kube_pod_status_phase{cluster=~"$cluster", namespace=~"$namespace", phase="Failed"} * 4) +
+            sum by (namespace, pod) (kube_pod_status_phase{cluster=~"$cluster", namespace=~"$namespace", pod=~"$pod", phase="Pending"} * 5)
```
Comment (davidsuba) [inline jsonnet/dashboards/k8s/pod-overview.libsonnet:96]: I would add `pod` to each part of this expression.

File: jsonnet/dashboards/k8s/container-overview.libsonnet (hunk 2 above)
Comment (michal-gubricky) [reply jsonnet/dashboards/k8s/container-overview.libsonnet:118]: When you add `container=~"$container"` to the query expressions and select all from drop-down menu you will get this type of error: `Unknown error during query transaction. Please check JS console logs.` This error do not show up when you select specific namespace for example.

File: jsonnet/dashboards/k8s/pod-overview.libsonnet (hunk 4 above)
Comment (michal-gubricky) [reply jsonnet/dashboards/k8s/pod-overview.libsonnet:96]: same issue as in the  `container-overview.libsonnet ` commented above [https://github.com/dNationCloud/kubernetes-monitoring/pull/58#discussion_r578268066](url)

File: jsonnet/dashboards/k8s/container-overview.libsonnet (hunk 2 above)
Comment (davidsuba) [reply jsonnet/dashboards/k8s/container-overview.libsonnet:118]: it seems like a grafana bug

File: jsonnet/dashboards/k8s/pod-overview.libsonnet (hunk 4 above)
Comment (davidsuba) [reply jsonnet/dashboards/k8s/pod-overview.libsonnet:96]: probably grafana bug

**Review by davidsuba — APPROVED** [review]

**Pull request merged**